### PR TITLE
SIMSBIOHUB-223: Implement UI improvements + changes for observations table

### PR DESCRIPTION
**Pull request by curtisupshall**
## Description of Changes

Implements UI improvements + changes for observations table.

### CODE REVIEW

File: api/src/repositories/observation-repository.ts [hunk 1]
```diff
@@ -141,7 +141,12 @@ export class ObservationRepository extends BaseRepository {
         longitude = EXCLUDED.longitude
     `);
 
-    sqlStatement.append(`RETURNING *;`);
+    sqlStatement.append(`
+      RETURNING
+        *,
+        latitude::double precision,
```
Comment (al-rosenthal) [inline api/src/repositories/observation-repository.ts:147]: working on the add/ edit for observations, I put a parse function in the `db.ts` that looks for the numeric type from postgres and converts it using `parseFloat`. It also means that whenever we need to add the lat/ long from this table we don't need to explicitly convert them, it is handled automatically.

**Review by al-rosenthal — COMMENTED** [review]
I'm able to put in a negative count for an animal

File: api/src/repositories/observation-repository.ts (hunk 1 above)
Comment (curtisupshall) [reply api/src/repositories/observation-repository.ts:147]: For now, let's leave this conversion in; Currently, updating the table is broken in dev. Assuming this PR will get merged before yours, we can remove this conversion in the subsequent PR.

File: app/src/features/surveys/observations/ObservationsTable.tsx [hunk 2]
```diff
@@ -14,6 +16,8 @@ const ObservationsTable = () => {
       flex: 1,
       minWidth: 250,
       disableColumnMenu: true,
+
+      // TODO: To be addressed by https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-288
```
Comment (al-rosenthal) [inline app/src/features/surveys/observations/ObservationsTable.tsx:20]: 👍

**Review by al-rosenthal — APPROVED** [review]
ran this locally, it looks good 🌦️

**Review by NickPhura — APPROVED** [review]

**Pull request merged**